### ISSUE TITLE
Allow group headings in section navigation

### ISSUE THREAD

**Issue by jrbarnes9** [ONSdigital/design-system#2621]
Currently, to produce the following lefthand navigation menu, the section navigation component has to be used multiple times for each sub-headed group. This results in multiple `nav` elements for what is essentially a single navigation menu.

This fails WCAG 2.1 Level A criterion 1.3.1 ('Info and relationships') because the `h3` group headings aren't contained within any landmarks.

Allowing multiple lists within the component, with sub-headings, will simplify and improve the labelling of the `nav` landmarks on the page.

***Note:*** 
We're already handling these groups/sections in this way in the [sub-navigation macro for the header component](https://github.com/ONSdigital/design-system/blob/a9ae208f024a34b24694c7ddf58994fafb2b69c9/src/components/navigation/_macro.njk#L86)

![Screenshot 2023-02-20 at 12 06 39](https://user-images.githubusercontent.com/43346934/220122994-e569849c-eb89-4625-92d7-f7568700ded4.png)

_Originally posted by @jrbarnes9 in https://github.com/ONSdigital/design-system/discussions/174#discussioncomment-5053551_

**Comment by yatesn** [comment]
Links to #242 in Service Manual
